.github/workflows/release_tests_fast.yml

```diff
@@ -0,0 +1,386 @@
+name: (Release) Fast GPU Tests on main
```

Comment:
Perhaps we could just add a comment explaining the reason behind why we're maintaining a duplicate workflow? I think it would make sense in having that comment because if we update anything in the `.github/workflows/push_tests.yml` workflow we might have to update that in the `.github/workflows/release_tests_fast.yml` as well.